### ISSUE TITLE
Uncaught ReferenceError: ActiveAdmin is not defined

### ISSUE THREAD

**Issue by 7vs** [activeadmin/activeadmin#7433]
In the browser console I get:
```
custom.js:64 Uncaught ReferenceError: ActiveAdmin is not defined
    at HTMLParagraphElement.<anonymous> (custom.js:64:1)
    at HTMLDocument.dispatch (jquery.js:4670:1)
    at HTMLDocument.elemData.handle (jquery.js:4490:1)
(anonymous) @ custom.js:64
dispatch @ jquery.js:4670
elemData.handle @ jquery.js:4490
```

I'm using webpacker and I can't figure out what's wrong with my setup.

For the look of the compiled active_admin.js file the active_admin js code is loaded after our custom code.

I'm trying to get an old code to work with no luck -although I can see the js code in the page, please see the screenshot.
![Screenshot 2022-04-06 at 16 28 43](https://user-images.githubusercontent.com/6040881/162019823-c5c7e60f-255d-437f-aee2-70e24b4b0dd6.png)

JQuery runs perfectly btw.

The `app/javascript/packs/active_admin.js` is set as:
```
// Load Active Admin's styles into Webpacker,
// see `active_admin.scss` for customization.
import "../stylesheets/active_admin";

const images = require.context("../images", true)
const imagePath = name => images(name, true)

import "@activeadmin/activeadmin";
import "@fortawesome/fontawesome-free/css/all.css";
import "arctic_admin";
import "activeadmin_addons";

require("trix")
require("@rails/actiontext")
import "../controllers"

// This is the custom JQuery code
import "./active_admin/custom"
import "./active_admin/dependent_select"
```

This is my `package.json` file:
```
{
  "name": "template-app",
  "private": true,
  "dependencies": {
    "@activeadmin/activeadmin": "^2.11.1",
    "@fortawesome/fontawesome-free": "^6.1.1",
    "@hotwired/stimulus": "^3.0.1",
    "@rails/actioncable": "^6.0.0",
    "@rails/actiontext": "^6.1.4-6",
    "@rails/activestorage": "^6.0.0",
    "@rails/ujs": "^6.0.0",
    "@rails/webpacker": "^5.4.3",
    "activeadmin_addons": "^1.0.0",
    "arctic_admin": "^4.0.2",
    "flatpickr": "^4.6.11",
    "jquery": "^3.6.0",
    "stimulus-flatpickr": "3.0.0-0",
    "trix": "^1.2.0",
    "turbolinks": "^5.2.0",
    "webpack": "^4.46.0",
    "webpack-cli": "^3.3.12"
  },
  "version": "0.1.0",
  "devDependencies": {
    "webpack-dev-server": "^3"
  }
}
```
Any help would be really appreciated

**Comment by 7vs** [comment]
With this line in `app/javascript/packs/active_admin.js`:
```
import "@activeadmin/activeadmin";
```
and your instructions for batch actions:
```
if $('body.admin_users').length
  $('a[data-prompt]').click ->
    ActiveAdmin.ModalDialog $(@).data('prompt'), comment: 'textarea',
      (inputs)=>
        $.post "/admin/users/#{$(@).data 'id'}/change_state",
          comment: inputs.comment, state: $(@).data('state'),
          success: ->
            window.location.reload()
```
I'm assuming I can have access to ActiveAdmin in the global state.

What am I missing or assuming wrong here?

Even when trying in the browser console ActiveAdmin is not defined

**Comment by 7vs** [comment]
Ok, sorted now. For whoever face same issue, here is what I had to do and my humble understanding of the issue.
I had to require the constant in the file where It's been used:
```
var ActiveAdmin = require("@activeadmin/activeadmin");
// plenty of JQuery code and some using the ActiveAdmin constant
...
```

I think that the file is treated as a local scope without access to the ActiveAdmin constant as webpacker exports isolated modules.
And even having the `import "@activeadmin/activeadmin"; ` make you think that it should make the constant available in that scope, is not what is happening, as I had the code pasted in the `app/javascript/packs/active_admin.js` after the import and it didn't work either.
